hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1704,34 +1745,39 @@ int BleObject::ConnectionsManager::disconnectAll() {
 }
 
 int BleObject::ConnectionsManager::updateConnectionParams(hal_ble_conn_handle_t connHandle, const hal_ble_conn_params_t* params) {
-    int ret;
-    if (fetchConnection(connHandle) == nullptr) {
-        return SYSTEM_ERROR_NOT_FOUND;
-    }
-    if (os_semaphore_create(&connParamsUpdateSemaphore_, 1, 0)) {
-        LOG(ERROR, "os_semaphore_create() failed");
-        return SYSTEM_ERROR_INTERNAL;
-    }
+    BleConnection* connection = fetchConnection(connHandle);
+    CHECK_TRUE(connection, SYSTEM_ERROR_NOT_FOUND);
     SCOPE_GUARD ({
-        connParamUpdateHandle_ = BLE_INVALID_CONN_HANDLE;
-        auto sem = connParamsUpdateSemaphore_;
-        connParamsUpdateSemaphore_ = nullptr;
-        os_semaphore_destroy(sem);
+        periphConnParamUpdateHandle_ = BLE_INVALID_CONN_HANDLE;
+        centralConnParamUpdateHandle_ = BLE_INVALID_CONN_HANDLE;
     });
+    if (connection->role == BLE_ROLE_PERIPHERAL) {
+        // If the automatic peripheral connection parameters update procedure is started.
+        if (!os_timer_is_active(connParamsUpdateTimer_, nullptr)) {
+            periphConnParamUpdateHandle_ = BLE_INVALID_CONN_HANDLE;
+            connParamsUpdateAttempts_ = 0;
```

Comment:
Should this be `volatile`? I see it modified in interrupt context, from the timer callback and from normal thread context.